crates/chain-state/src/in_memory.rs

```diff
@@ -112,7 +112,12 @@ impl InMemoryState {
     /// Returns the pending state corresponding to the current head plus one,
     /// from the payload received in newPayload that does not have a FCU yet.
     pub(crate) fn pending_state(&self) -> Option<Arc<BlockState>> {
-        self.pending.borrow().as_ref().map(|state| Arc::new(BlockState::new(state.block.clone())))
+        self.pending.borrow().as_ref().map(|state| {
+            Arc::new(BlockState::with_parent(
+                state.block.clone(),
+                state.parent.clone().map(|p| (*p).clone()),
+            ))
```

Comment:
Maybe I missed something here but it seems normal to have the information about the parent state here when we call the `pending_state` functions rather than a `None` resulting from the previous `BlockState::new` call.

Reply:
this is actually a bug here because this should always clone the entire pending state so that it doesn't lose the parent chain context